front/src/common/Map/Search/MapSearchSignal.tsx

```diff
@@ -25,15 +26,44 @@ export type SortType = {
   asc: boolean;
 };
 
-const SIGNAL_ASPECTS = signalAspects;
-
 const MapSearchSignal = ({ updateExtViewport, closeMapSearchPopUp }: MapSearchSignalProps) => {
   const map = useSelector(getMap);
   const infraID = useSelector(getInfraID);
+  const { t } = useTranslation(['translation', 'map-search']);
+
+  const signalSystems = {
+    ALL: t('map-search:all'),
+    BAL: 'BAL',
+    BAPR: 'BAPR',
+    TVM: 'TVM',
+  };
+
+  const signalSettingsMap = {
+    [signalSystems.ALL]: [],
+    [signalSystems.BAL]: ['Nf'],
+    [signalSystems.BAPR]: ['Nf', 'distant'],
+    [signalSystems.TVM]: ['is_430'],
+  };
+
+  const signalSettingsDisplay: { [key: string]: string } = {
+    Nf: t('map-search:signalSettings.Nf'),
+    distant: t('map-search:signalSettings.distant'),
+    is_430: t('map-search:signalSettings.is_430'),
+  };
+
+  const reversedSignalSettingsDisplay = Object.entries(signalSettingsDisplay).reduce(
+    (acc, [key, value]) => {
+      acc[value] = key;
+      return acc;
+    },
+    {} as { [key: string]: string }
+  );
```

Comment:
If this is only const, can you move it outside the component ? (above const MapSearchSignal, l29). Otherwise it will be re-instantiated at each re-render 

Also we use capital letters for const and enums: SIGNALING_SYSTEMS, SIGNAL_SETTINGS_MAP, SIGNAL_SETTINGS_DISPLAY
```suggestion
  const signalSystems = {
    ALL: t('map-search:all'),
    BAL: 'BAL',
    BAPR: 'BAPR',
    TVM: 'TVM',
  };

  const signalSettingsMap = {
    [signalSystems.ALL]: [],
    [signalSystems.BAL]: ['Nf'],
    [signalSystems.BAPR]: ['Nf', 'distant'],
    [signalSystems.TVM]: ['is_430'],
  };

  const signalSettingsDisplay: { [key: string]: string } = {
    Nf: t('map-search:signalSettings.Nf'),
    distant: t('map-search:signalSettings.distant'),
    is_430: t('map-search:signalSettings.is_430'),
  };

  const reversedSignalSettingsDisplay = Object.entries(signalSettingsDisplay).reduce(
    (acc, [key, value]) => {
      acc[value] = key;
      return acc;
    },
    {} as { [key: string]: string }
  );
```